### DISCUSSION TITLE
Saving Images

### DISCUSSION THREAD

**Discussion by makleo2002** [clientIO/joint#2642] [Q&A] [answered]
### Introduction

how do I save an image from paper in SVG JPG PNG formats and then download it?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You are looking for [format.Raster](https://resources.jointjs.com/docs/rappid/v4.0/format.Raster.html) and [format.SVG](https://resources.jointjs.com/docs/rappid/v4.0/format.SVG.html) plugins.

There are several examples of how to export a diagram to an image: https://www.jointjs.com/demos?feature=Export+to+PNG%2FJPEG

And here's a simple example:
```ts
import { format, util } from '@joint/plus';

/*... */

format.toSVG(
    paper,
    (svg) => {
        util.downloadDataUri(
            `data:image/svg+xml,${encodeURIComponent(svg)}`,
            'joint-plus.svg'
        );
    },
    { useComputedStyles: false }
);
```
